bundles/org.openhab.binding.ventaair/src/main/java/org/openhab/binding/ventaair/internal/Communicator.java

```diff
@@ -0,0 +1,179 @@
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.math.BigDecimal;
+import java.net.Socket;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ventaair.internal.VentaThingHandler.StateUpdatedCallback;
+import org.openhab.binding.ventaair.internal.message.CommandMessage;
+import org.openhab.binding.ventaair.internal.message.DeviceInfoMessage;
+import org.openhab.binding.ventaair.internal.message.Header;
+import org.openhab.binding.ventaair.internal.message.Message;
+import org.openhab.binding.ventaair.internal.message.action.Action;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link Communicator} is responsible for sending/receiving commands to/from the device
+ *
+ * @author Stefan Triller - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class Communicator {
+    private final Duration COMMUNICATION_TIMEOUT = Duration.ofSeconds(5);
+
+    private final Logger logger = LoggerFactory.getLogger(Communicator.class);
+
+    private @Nullable String ipAddress;
+    private Header header;
+    private int pollingTimeInSeconds;
+    private StateUpdatedCallback callback;
+
+    private Gson gson = new Gson();
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public Communicator(@Nullable String ipAddress, Header header, @Nullable BigDecimal pollingTime,
+            StateUpdatedCallback callback) {
+        this.ipAddress = ipAddress;
+        this.header = header;
+        if (pollingTime != null) {
+            this.pollingTimeInSeconds = pollingTime.intValue();
+        } else {
+            this.pollingTimeInSeconds = 60;
+        }
+        this.callback = callback;
+    }
+
+    /**
+     * Sends a request message to the device, reads the reply and informs the listener about the current device data
+     */
+    public void pollDataFromDevice() {
+        String messageJson = gson.toJson(new Message(header));
+
+        logger.debug("Opening connection to device for polling");
+
+        try (Socket socket = new Socket(ipAddress, VentaAirBindingConstants.PORT)) {
+            socket.setSoTimeout((int) COMMUNICATION_TIMEOUT.toMillis());
+            InputStream input = socket.getInputStream();
+            OutputStream output = socket.getOutputStream();
+
+            byte[] dataToSend = buildMessageBytes(messageJson, "GET", "Complete");
+            logger.debug("Sending request data message (String):\n{}", new String(dataToSend));
+            logger.debug("Sending request data message (bytes): [{}]", HexUtils.bytesToHex(dataToSend, ", "));
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging